src/main/scala/fi/vm/sade/hakurekisteri/integration/VirkailijaRestClient.scala

```diff
@@ -263,12 +298,20 @@ object JsonExtractor extends HakurekisteriJsonSupport {
     }
     new FunctionHandler[T](f) {
       override def onCompleted(response: Res): T = {
-        if (codes.contains(response.getStatusCode))
+        if (codes.contains(response.getStatusCode) && !isRedirectToCasLogin(response))
```

Comment:
Jos haluaa vähän vähentää if-ehtoja, se onnistunee, jos laittaa ekaksi tuon `if (isRedirectToCasLogin(response))` -käsittelyn. Vaikka toisaalta siinäkin on pointtia, että happy path on ekana ja poikkeustapaukset vasta sen jälkeen.

Reply:
If-ehdoissa ehkä kuitenkin less is more